velox/functions/prestosql/aggregates/QDigestAggAggregate.cpp

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/prestosql/aggregates/QDigestAggAggregate.h"
+
+#include "velox/exec/Aggregate.h"
+#include "velox/exec/SimpleAggregateAdapter.h"
+#include "velox/expression/FunctionSignature.h"
+#include "velox/functions/lib/QuantileDigest.h"
+#include "velox/functions/prestosql/aggregates/AggregateNames.h"
+#include "velox/functions/prestosql/types/QDigestRegistration.h"
+
+using namespace facebook::velox::exec;
+using namespace facebook::velox::functions::qdigest;
+
+namespace facebook::velox::aggregate::prestosql {
+
+namespace {
+
+template <typename T, typename Func, bool DefaultMaxError>
+struct QDigestAccumulator {
+  QuantileDigest<T> digest;
+
+  QDigestAccumulator() = delete;
+
+  explicit QDigestAccumulator(HashStringAllocator* allocator, Func* /*fn*/)
+      : digest(
+            StlAllocator<T>(allocator),
+            DefaultMaxError ? 0.01 : kUninitializedMaxError) {}
+
+  void addInput(HashStringAllocator* /*allocator*/, exec::arg_type<T> value) {
+    digest.add(value, 1);
```

Comment:
Thoughts on using 1 as default in QuantileDigest library? Currently, there is no other usage but if we create construct_qdigest, similar to constuct_tdigest for eg.